pkgs/development/tools/jq/default.nix

```diff
@@ -12,7 +13,17 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "doc" "man" "dev" "lib" "out" ];
 
-  buildInputs = [ oniguruma ];
+  # inspired by app-misc/jq/jq-1.6-r3.ebuild from Gentoo
+  postPatch = ''
+    rm -fr ./modules/oniguruma
+  '' + lib.optionalString (!onigurumaSupport) ''
+    sed -e '/modules\/oniguruma/d' -i Makefile.am
+    sed -e 's,tests/onigtest,,' -i Makefile.am
+  '';
+
+  buildInputs = [ ]
+    ++ lib.optional onigurumaSupport [ oniguruma ]
+    ++ lib.optional (!onigurumaSupport) [ autoreconfHook ];
```

Comment:
Why not use configure flags? It seems `--with-onigurama=no` should be supported:

https://github.com/stedolan/jq/blob/master/configure.ac#L245

Reply:
Well, I can use configure flags on top of this for extra clarity, but this is essential anyway:
If oniguramaSupport = false, onigurma must not be in builtInputs and autoreconfHooks is needed.

Reply:
I tested this a bit, and indeed only adding `--with-oniguruma=no` and not adding it to the inputs isn't enough, the `onigtest` test is still built and it fails:

```
jq> PASS: tests/optionaltest
jq> PASS: tests/mantest
jq> PASS: tests/jqtest
jq> FAIL: tests/onigtest
jq> PASS: tests/shtest
jq> PASS: tests/utf8test
jq> PASS: tests/base64test
```

This should be reported upstream IMO, such a build configuration should not require altering `Makefile.am` like you do, in order to avoid this test.

Also, please note that in anycase `autoreconfHook` should be added to native build inputs.

Reply:
I agree that unconditional build of oniguma test is bug. I am going to prepare patch, so eventually it will get fixed upstream. In mean time, moved autoreconfHook into native build inputs.

Reply:
Here is upstream PR: https://github.com/stedolan/jq/pull/2292

Reply:
> Here is upstream PR: [stedolan/jq#2292](https://github.com/stedolan/jq/pull/2292)

Thanks for writing the patch @KAction . Perhaps now it'd be a good idea now to switch to `fetchFromGitHub`, use `fetchpatch` to apply your patch and still use `autoreconfHook`. This way, the next update will require removing the patch because it won't apply (hoping upstream will merge your PR). The `sed` commands we have currently may bit rot in the long future.